### DISCUSSION TITLE
STM32F407VGT6 design guidance

### DISCUSSION THREAD

**Discussion by ITstreet1** [stm32duino/Arduino_Core_STM32#2197] [Variants]
Hi guys,

I need a word or two about this MCU. 
What USART pins should I use to upload a sketch over a serial adapter? Usually, these are PA9 and PA10. But, according to the variants:
https://github.com/stm32duino/Arduino_Core_STM32/blob/main/variants/STM32F4xx/F407V(E-G)T_F417V(E-G)T/variant_generic.h
It should be PA0 and PA1. Or I am wrong...

This is the schematic I have done so far:

![Schematic](https://github.com/stm32duino/Arduino_Core_STM32/assets/30090189/74576a4b-3aea-4a3d-85ad-fbc0ec5c8e9f)

Any suggestion?

**Comment by fpistm** [comment]
Hi @ITstreet1 
In fact there is no link between the U(S)ART pins defined in the variant and the one used to flash.
From the variant, it defines the pins used for default Serial instance.
To flash you have to refers to this documentation:
https://www.st.com/resource/en/application_note/an2606-stm32-microcontroller-system-memory-boot-mode-stmicroelectronics.pdf

So in your case:
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20641798/648179c2-5591-4bc6-8d7c-852e7325da55)

This is the bootloader which will be in charge to properly init the Serial port. See the pattern 1 to properly enter in BL mode:
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20641798/0859ea47-8dcc-4a9a-a057-48cbd2902db5)

**Comment by ITstreet1** [comment]
Hi @fpistm 
Not sure how I managed to open two topics for the same thing. LOL. Please merge it if you can.

The idea is to make a design with an STM32 and an Ethernet. And to be able to program it using Arduino IDE, but also with the Necto Studio.

MikroElektronika has a working SiBrain design for a similar MCU.
https://download.mikroe.com/examples/full-featured-boards/8th-generation/schematic/SiBRAIN/SiBRAIN-for-STM32F407ZG-schematic.pdf

This is what I have come up with so far.

![Sheet_1](https://github.com/stm32duino/Arduino_Core_STM32/assets/30090189/ce31bfc4-fbbb-4029-9a0c-06839e0842e2)

![Sheet_2](https://github.com/stm32duino/Arduino_Core_STM32/assets/30090189/dd6cf7bc-d8db-4944-8852-5c5fa8e9bb44)

If you have a minute or two to take a look at this one, with be nice.

**Reply by fpistm** [reply]
Unfortunately I could not help as I'm mainly on software side. You could try on the forum stm32duino.com or on st community.

**Comment by fpistm** [comment]
Link to https://github.com/orgs/stm32duino/discussions/2195
